targets/TARGET_STM/TARGET_STM32H7/analogin_device.c

```diff
@@ -67,6 +67,9 @@ void analogin_init(analogin_t *obj, PinName pin)
         // Get the functions (adc channel) from the pin and assign it to the object
         function = pinmap_function(pin, PinMap_ADC);
         // Configure GPIO
+#if defined(ALTC)
+    if ((pin != PA_0C) && (pin != PA_1C) && (pin != PC_2C) && (pin != PC_3C))
+#endif
```

Comment:
This will stop analog support of arduino Ax pins ?

https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_DISCO_H747I/PinNames.h#L308-L311

Reply:
No, analog Ax pins are working also after the change. On our board we are using `PA_0C` `PA_1C` `PC_2C` `PC_3C` `PC2` `PC3` `PA4` and `PA6` as analog input pins and they are all working. It's my understanding that there is no need (and it is in some way wrong) to call the `pinmap_pinout` functions for PX_XC because they cannot be configured as all the others GPIO. `pinmap_pinout` function is used to change GPIO port configuration (pull-up, pull-down, speed, input output...) but this pins don't have these options.

Reply:
OK, it's working because in analogin, pinmap_pinout sets pin to analog and PullNone, which is the default pin state...

But I don't like the current correction :-)
I am thinking about a better patch in pinmap.c

Reply:
Could you have a look on https://github.com/jeromecoutant/mbed/commit/ca862ed7a51a994805be7ffebadd57006bf0f173